### ISSUE TITLE
FGHJ

### ISSUE THREAD

**Issue by darewolf007** [darewolf007/Python_lib#3]
def farthest_point_sample(xyz, npoint):
    """
    Input:
        xyz: pointcloud data, [B, N, 3]
        npoint: number of samples
    Return:
        centroids: sampled pointcloud index, [B, npoint]
    """
    device = xyz.device
    B, N, C = xyz.shape
    centroids = torch.zeros(B, npoint, dtype=torch.long).to(device)
    distance = torch.ones(B, N).to(device) * 1e10
    farthest = torch.randint(0, N, (B,), dtype=torch.long).to(device)
    batch_indices = torch.arange(B, dtype=torch.long).to(device)
    for i in range(npoint):
        centroids[:, i] = farthest
        centroid = xyz[batch_indices, farthest, :].view(B, 1, 3)
        dist = torch.sum((xyz - centroid) ** 2, -1)
        # print('dist',dist.shape)
        mask = dist < distance
        # print('mask',mask.shape,mask)
        distance[mask] = dist[mask]
        # print('distance',distance.shape,distance)
        farthest = torch.max(distance, -1)[1]
    return centroids
